src/events/neo4j.js

```diff
@@ -0,0 +1,265 @@
+const tracer = require('../tracer.js');
+const errorCode = require('../proto/error_code_pb.js');
+
+/**
+ * Extracts the relevant arguments from provided arguments
+ * @returns {Object} Object of the extracted arguments
+ */
+function getArgsFromFunction(...args) {
+    return {
+        query: args[0],
+        params: args[1],
+    };
+}
+
+
+/**
+ * Gets session's address info from Transaction/Session instance
+ * @param {neo4j.Session|neo4j.Transaction} session A DB session (could be Session of Transaction)
+ * @returns {Object} json with address info { host, port }
+ */
+function getAddressInfo(session) {
+    let port = '7474';
+    let host = 'neo4j';
+    let connectionHolder;
+
+    switch (session.constructor.name) {
+    // By default, Session opens two same connectionProviders - Read and Write, and not default one
+    case 'Session':
+        connectionHolder = session._readConnectionHolder;
+        break;
+
+    case 'Transaction':
+        connectionHolder = session._connectionHolder;
+        break;
+
+    default:
+        return {
+            host, port,
+        };
+    }
+
+    if (session && connectionHolder && connectionHolder._connectionProvider) {
+        const connectionProvider = connectionHolder._connectionProvider;
+        port = connectionProvider._address.port();
+        host = connectionProvider._address.host();
+    }
+
+    return {
+        port, host,
+    };
+}
+
+
+/**
+ * Parse from DB session with Neo4j server, useful metadata
+ * @param {neo4j.Session|neo4j.Transaction} session A DB session (could be Session of Transaction)
+ * @returns {{metadata: Object, fullMetadata: Object}} JSONs with command's metadata and command's
+ *      full metadata which contains additional data about the event
+ */
+function getNeo4JSessionMetadata(session) {
+    const metadata = {};
+    const fullMetadata = {};
+
+    let connectionHolder;
+    let sessionMode;
+    let operationMode = '';
+
+    switch (session.constructor.name) {
+    // By default, Session opens two same connectionProviders - Read and Write, and not default one
+    case 'Session':
+        connectionHolder = session._readConnectionHolder;
+        sessionMode = 'READ|WRITE';
+        break;
+
+    case 'Transaction':
+        connectionHolder = session._connectionHolder;
+        sessionMode = connectionHolder._mode;
+
+        break;
+
+    default:
+        return;
+    }
+
+    switch (sessionMode) {
+    case 'READ':
+        operationMode = 'Read';
+        break;
+
+    case 'WRITE':
+        operationMode = 'Write';
+        break;
+
+    case 'READ|WRITE':
+        operationMode = 'Read|Write';
+        break;
+
+    default:
+        break;
+    }
+
+    metadata.dbName = connectionHolder._database ? connectionHolder._database : 'default';
+    metadata.operation = `${session.constructor.name}:${operationMode}`;
+
+    fullMetadata.config = connectionHolder._connectionProvider._config;
```

Comment:
add safe checks

Reply:
Fixed? (does that fix also fixed the prev comment?)